src/librustdoc/html/format.rs

```diff
@@ -207,35 +207,27 @@ impl clean::GenericParamDef {
                 f.write_str(self.name.as_str())?;
 
                 if !bounds.is_empty() {
-                    if f.alternate() {
-                        write!(f, ": {:#}", print_generic_bounds(bounds, cx))?;
-                    } else {
-                        write!(f, ": {}", print_generic_bounds(bounds, cx))?;
-                    }
+                    f.write_str(": ")?;
+                    fmt::Display::fmt(&print_generic_bounds(bounds, cx), f)?;
                 }
 
                 if let Some(ref ty) = default {
-                    if f.alternate() {
-                        write!(f, " = {:#}", ty.print(cx))?;
-                    } else {
-                        write!(f, " = {}", ty.print(cx))?;
-                    }
+                    f.write_str(" = ")?;
+                    fmt::Display::fmt(&ty.print(cx), f)?;
                 }
 
                 Ok(())
             }
             clean::GenericParamDefKind::Const { ty, default, .. } => {
-                if f.alternate() {
-                    write!(f, "const {}: {:#}", self.name, ty.print(cx))?;
-                } else {
-                    write!(f, "const {}: {}", self.name, ty.print(cx))?;
-                }
+                write!(f, "const {}: ", self.name)?;
+                fmt::Display::fmt(&ty.print(cx), f)?;
 
                 if let Some(default) = default {
+                    f.write_str(" = ")?;
                     if f.alternate() {
-                        write!(f, " = {default:#}")?;
```

Comment:
Previously this was formatting a `String` with `{:#}` but `#` doesn't have any effect on `<String as Display>`. Furthermore this didn't escape const generic defaults.

Reply:
Here's a nice picture cuz why not:

![Screenshot 2024-02-15 at 05-23-51 S in unesc - Rust](https://github.com/rust-lang/rust/assets/14913065/6533bceb-8702-4e82-83fb-fb721e3bb6b0)

Reply:
Mouahaha. Nice one. :D